pymc/model/core.py

```diff
@@ -1584,6 +1584,36 @@ def __getitem__(self, key):
     def __contains__(self, key):
         return key in self.named_vars or self.name_for(key) in self.named_vars
 
+    def __copy__(self):
+        """
+        Clone a pymc model by overiding the python copy method using the clone_model method from fgraph.
+        if guassian process variables are detected then an exception will be raised.
+        """
+        from pymc.model.fgraph import clone_model
+
+        check_for_gp_vars = [
+            k for x in ["_rotated_", "_hsgp_coeffs_"] for k in self.named_vars.keys() if x in k
+        ]
+        if len(check_for_gp_vars) > 0:
+            raise Exception("Unable to clone Gaussian Process Variables")
```

Comment:
This should be in the low-level utility used by `clone_model`, `fgraph_from_model`? Also I think it should be a warning, because we are not sure something with `_rotated_` is a GP or just a user variable name for a vanilla variable.

Also @bwengals any other names we could look for to detect GPs in the model? This is not perfect, but avoiding as much surprise as possible would be great.

Reply:
@ricardoV94 Yes, you are absolutely correct. This should be a warning and yes, I can move it to the lower-level utility used by `clone_model`. Do you know if there is a way that I can check model variables to discern the type of the variable it is? Something like `type(model.named_vars)` and it would say it was a GP

Reply:
Nope, no way to know it was a GP, that's the unfortunate bit. GPs produce vanilla MvNormals or deterministics

Reply:
Sorry for the very delayed response.  No, there isn't.  What I've done to smuggle GPs out with the model in the past is to do something like:

```python
with pm.Model() as model:
    # model code here
    gp1 = pm.gp.Latent(...)
    gp2  = pm.gp.HSGP(...)

model.gps = {"gp1": gp1, "gp2": gp2, ...}
```

If GPs added themselves to the model context automatically they could be tracked.  They don't have names though, so need a little thought for how to key that dictionary, although maybe putting them in a list is OK.

Reply:
Hey @bwengals, if I understand you correctly, you are saying that we can add GPs to the Model class as an attribute and that would happen in `core.py` in the section after line 539 and then we can pull those variables just like the rest of the variables in `fgraph_from_model()` in the section after line 169 inside of the `fgraph.py` file. 
(sorry for the line number references I couldn't figure out how to highlight the code in the comment. I recently just started contributing to open source)